cleanup: Fixing some grammar and wording.

larrydewey opened the pull request:
This adds some cleanup for the existing documentation, adds some language specifiers for code blocks, as well as some fixes for minor spelling issues.

larrydewey committed:
cleanup: Fixing some grammar and wording.

This adds some cleanup for the existing documentation, adds some
language specifiers for code blocks, as well as some fixes for minor
spelling issues.

Signed-off-by: Larry Dewey <larry.dewey@amd.com>

fitzthum reviewed (approved):
LGTM. Thanks!

dcmiddle commented on guides/sev.md:
```diff
@@ -58,20 +58,20 @@ sudo ./sevctl/target/debug/sevctl export --full /opt/sev/cert_chain.cert
 The SEV policy can also be set by adding `io.katacontainers.config.sev.policy: "<SEV POLICY>"` to your pod configuration.
 Setting the second bit of the policy enables SEV-ES.
-For more information see chapter 3 of the AMD Secure Encrypted Virtualization API.
+For more information see chapter 3 of the [Secure Encrypted Virtualization API](https://www.amd.com/system/files/TechDocs/55766_SEV-KM_API_Specification.pdf#page=31).
```
I haven't seen a license like this on other things in open source. Can this be updated to something more open source conventional?

dcmiddle replied:
It's not that the spec needs to be "open source" it's just that if there's something like a paywall in front of a spec it's not friendly for developers to access. In this case, there's an unusual license to accept to look at the document. Maybe there's a friendlier API specification to point to rather than that one?

dcmiddle replied:
Maybe y'all could republish it with an open source friendly license? I think most specs just have copyright and some disclaimer. (not a lawyer)

magowan replied:
So I can follow the link https://www.amd.com/system/files/TechDocs/55766_SEV-KM_API_Specification.pdf#page=31 without needing to accept anything?
Though the first page of the spec which requires the reader to acknowledge a specification agreement before reading further concerns me? (Though I was only checking the link not going on to read the document anyway)

I do share @dcmiddle license concerns, But if there was a requirement to acknowledge something in order to download (more a friendly to use concern)  -  Now there isn' acknowledgement to download, but there is still an specification agreement on the first page that perhaps worries me more? Is there a risk that someone working on other TEE's could inadvertently click on link from CoCo docs and end up with a specification document that could cause legal headaches?

magowan replied:
Perhaps remove link, lets merge this PR and put the question to the community regarding concerns on accessing specs?

dcmiddle replied:
That makes sense to me. The rest of the PR is great and it would be better to see it merged than waiting indefinitely on this point. (Unless Larry is close to resolution on the licensing).

fitzthum replied:
fwiw I really see this as a non-issue. I think the legalese is pretty standard. Our work doesn't come close to violating any of the clauses and people who click on the link can decide for themselves if they want to read the spec. I think there is basically no chance that AMD will adjust the license. If we can't get past this then I guess removing the link is an easy option, but I think there is really nothing to worry about.

dcmiddle reviewed (commented):
Thanks for the fix-ups. Generally looks good. One question on licensing reference.

larrydewey committed:
docs: Removing link for now

Due to some of the discussions in the community, we are removing this
for now.

Signed-off-by: Larry Dewey <larry.dewey@amd.com>
